### PR TITLE
refactor: Remove usage of lazy_static! not neard files

### PR DESCRIPTION
**Pull request by pmnoxx**
Full change for removing `Lazy` and usage of `metrics::Result`.

See https://github.com/near/nearcore/issues/5317

I'm splitting this PR into smaller ones. I'll do more split as more RPs get merged.
- [x] 19 files chain/jsonrpc https://github.com/near/nearcore/pull/5323
- [x] 11 files chain #5318 
- [x] 10 files core only https://github.com/near/nearcore/pull/5366
- [x] 3 files runtime https://github.com/near/nearcore/pull/5367
- [x] 8 files other places #5321
- [x] Remove unused methods in metrics https://github.com/near/nearcore/pull/5374

### CODE REVIEW

File: core/store/src/db.rs [hunk 1]
```diff
@@ -193,54 +194,51 @@ impl DBCol {
...
-}
+
+pub static IS_COL_RC: Lazy<Vec<bool>> = Lazy::new(|| {
+    let mut col_rc = vec![false; NUM_COLS];
+    col_rc[DBCol::ColState as usize] = true;
+    col_rc[DBCol::ColTransactions as usize] = true;
+    col_rc[DBCol::ColReceipts as usize] = true;
+    col_rc[DBCol::ColReceiptIdToShardId as usize] = true;
+    col_rc
+});
```
Comment (matklad) [inline core/store/src/db.rs:241]: Unrelated to the PR, but I think all these don't need to be lazy, and can be just constexprs

**Review by matklad — APPROVED** [review]

File: core/store/src/db.rs (hunk 1 above)
Comment (pmnoxx) [reply core/store/src/db.rs]: We should make a separate PR to fix that.

**Comment by pmnoxx** [comment]
@matklad I'm in favor in general of splitting large changes into smaller into groups by code owners.

Big changes are good to be split into 5 groups, though maybe a better division can be found.
- chain/jsonrc
- chain/* not jsorc
- core
- runtime
- everything else


What do you think:

**Comment by matklad** [comment]
Yeah, smaller changes are always better!

**Review by frol — APPROVED** [review]

**Review by nikurt — APPROVED** [review]

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**